test/XRP/web-socket-network-client.test.ts

```diff
@@ -0,0 +1,123 @@
+import { assert } from 'chai'
+import { Wallet, XrplNetwork, XrpUtils } from 'xpring-common-js'
+import WebSocketNetworkClient from '../../src/XRP/network-clients/web-socket-network-client'
+import { WebSocketTransactionResponse } from '../../src/XRP/shared/rippled-web-socket-schema'
+import XrpError from '../../src/XRP/shared/xrp-error'
+import XrpClient from '../../src/XRP/xrp-client'
+
+import XRPTestUtils from './helpers/xrp-test-utils'
+
+// A timeout for these tests.
+// eslint-disable-next-line @typescript-eslint/no-magic-numbers -- 1 minute in milliseconds
+const timeoutMs = 60 * 1000
+
+const rippledGrpcUrl = 'test.xrp.xpring.io:50051'
+const rippledWebSocketUrl = 'wss://wss.test.xrp.xpring.io'
+
+const webSocketNetworkClient = new WebSocketNetworkClient(
+  rippledWebSocketUrl,
+  console.log,
+)
+
+function sleep(ms: number) {
+  return new Promise((resolve) => setTimeout(resolve, ms))
+}
+
+describe('WebSocket Tests', function (): void {
+  // Retry integration tests on failure.
+  this.retries(3)
+
+  // A Wallet with some balance on Testnet.
+  let wallet: Wallet
+  let wallet2: Wallet
+  before(async function () {
+    wallet = await XRPTestUtils.randomWalletFromFaucet()
+    wallet2 = await XRPTestUtils.randomWalletFromFaucet()
+  })
+
+  after(function (done) {
+    webSocketNetworkClient.close()
+    done()
+  })
+
+  it('subscribeToAccount - valid request', async function (): Promise<void> {
+    this.timeout(timeoutMs)
+
+    const xAddress = wallet.getAddress()
+    const classicAddress = XrpUtils.decodeXAddress(xAddress)
+    const address = classicAddress!.address
+
+    const xrpAmount = '100'
+    const subscriptionId = 'subscribe_transaction_' + address
+
+    let messageReceived = false
+    const callback = (data: WebSocketTransactionResponse) => {
+      messageReceived = true
+      assert.equal(data.engine_result, 'tesSUCCESS')
+      assert.equal(data.engine_result_code, 0)
+      assert.equal(
+        data.engine_result_message,
+        'The transaction was applied. Only final in a validated ledger.',
+      )
+      assert.equal(data.meta.TransactionResult, 'tesSUCCESS')
+      assert.equal(data.status, 'closed')
```

Comment:
a little confused by this `data.status` ?  What does `'closed'` mean?  I see here that the `status` field in the response should read `success` ?

Reply:
My understanding is that this refers to the status of the transaction/ledger referred to in this response

see https://xrpl.org/subscribe.html#transaction-streams

Reply:
hmm, yeah that's the only context in which I've really heard "closed" used as well.. and the reason that concerns me is that I think we should only be pulling data from *validated* ledgers - not open or closed...  also, the `data.validated` field is `true`... 

Looking at the documentation, I don't see any other options to pass into the request to specify validated-only ledgers, and the documentation indicates that subscribing to an account *should* only return data from validated ledgers.  This, combined with the fact that the `validated` field is indeed `true` makes me feel good enough about things... but it's a bit weird.  We can keep our radar out for this weirdness when we build our demo.

Reply:
Giving an `accounts` field means that we only get finalized transactions - if we used `accounts_proposed` instead, we'd also get not-finalized transactions